lab/src/Microsoft.ServiceModel.Syndication/src/Atom10FeedFormatter.cs

```diff
@@ -243,162 +223,281 @@ internal static void ReadCategory(XmlReader reader, SyndicationCategory category
                         ((IDisposable)extWriter).Dispose();
                     }
                 }
-                reader.ReadEndElement();
+
+                await reader.ReadEndElementAsync();
             }
             else
             {
-                reader.ReadStartElement();
+                await reader.ReadStartElementAsync();
             }
         }
 
-        internal static TextSyndicationContent ReadTextContentFrom(XmlReader reader, string context, bool preserveAttributeExtensions)
+        internal static async Task<TextSyndicationContent> ReadTextContentFromAsync(XmlReaderWrapper reader, string context, bool preserveAttributeExtensions)
         {
             string type = reader.GetAttribute(Atom10Constants.TypeTag);
-            return ReadTextContentFromHelper(reader, type, context, preserveAttributeExtensions);
+            return await ReadTextContentFromHelper(reader, type, context, preserveAttributeExtensions);
         }
```

Comment:
Remove async keyword from method and remove the await. Just return the Task from ReadTextContentFromHelper